### PR TITLE
DOC: Merge together DISTUTILS.rst.txt#template-files" and distutils.rst#conversion-of-src-files

### PR DESCRIPTION
**Pull request by ojaswi12**
I created a new PR with the correct commits.

### CODE REVIEW

**Comment by mattip** [comment]
You need to [rebase this off master](http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html?highlight=rebase#rebasing-on-master) to fix the CI build error which was fixed in #13206

**Comment by mattip** [comment]
Fixes #13129

File: doc/DISTUTILS.rst.txt [hunk 1]
```diff
@@ -361,6 +361,9 @@ during the build process are ready to be compiled. This form
 of generic typing is also supported for C header files (preprocessed to
 produce :file:`.h` files).
 
+Additionally, NumPy distutils supports automatic conversion of source files named *.src". This facility can be used to maintain very similar code blocks requiring only simple changes between blocks. During the build phase of setup, if a template file named <somefile>.src is encountered, a new file named <somefile> is constructed from the template and placed in the build directory to be used instead. Two forms of template conversion are supported. The first form occurs for files named <file>.ext.src where ext is a recognized Fortran extension (f, f90, f95, f77, for, ftn, pyf). The second form is used for all other cases. Read more `here`_.
+
+.. _here: http://www.numpy.org/devdocs/reference/distutils.html#conversion-of-src-files
```
Comment (mattip) [inline doc/DISTUTILS.rst.txt]: This is a good start. You should remove lines 303 - 363 from here and merge them with the content you linked too

**Comment by ojaswi12** [comment]
Sir the CI build merging is already updated into my branch. I don't think rebasing is needed as there are no differences between both repositories apart from this document.

File: doc/DISTUTILS.rst.txt [hunk 2]
```diff
@@ -361,6 +361,9 @@ during the build process are ready to be compiled. This form
 of generic typing is also supported for C header files (preprocessed to
 produce :file:`.h` files).
 
+Additionally, NumPy distutils supports automatic conversion of source files named *.src". This facility can be used to maintain very similar code blocks requiring only simple changes between blocks. During the build phase of setup, if a template file named <somefile>.src is encountered, a new file named <somefile> is constructed from the template and placed in the build directory to be used instead. Two forms of template conversion are supported. The first form occurs for files named <file>.ext.src where ext is a recognized Fortran extension (f, f90, f95, f77, for, ftn, pyf). The second form is used for all other cases. Read more `here`_.
```
Comment (mattip) [inline doc/DISTUTILS.rst.txt]: The `*.src"` is breaking the build. it should be enclosed in two back-slashes. Same with any non-text code like `<somefile>.src`, `<somefile>`, ...
```
``*.src``
```

**Comment by mattip** [comment]
> CI build merging is already updated into my branch

My mistake. The red X next to the commit indicates the build is broken, you should be able to click on it to see why.

In the future please use a useful commit message like `DOC: merge duplicate descriptions`

File: doc/DISTUTILS.rst.txt (hunk 1 above)
Comment (ojaswi12) [reply doc/DISTUTILS.rst.txt]: I removed lines 308 to 363, and at the end of the first paragraph I added a link to this page: http://www.numpy.org/devdocs/reference/distutils_guide.html#template-files. Is this what you wanted? I would add the file here but git is not allowing me to for some reason.

Comment (ojaswi12) [reply doc/DISTUTILS.rst.txt]: This is how it looks afterwards

image

File: doc/DISTUTILS.rst.txt (hunk 2 above)
Comment (ojaswi12) [reply doc/DISTUTILS.rst.txt]: I'll change it right now

File: doc/DISTUTILS.rst.txt (hunk 1 above)
Comment (mattip) [reply doc/DISTUTILS.rst.txt]: > I would add the file here but git is not allowing me to for some reason

What file are you adding? You should be modifying the file in-place inside your clone of a git repo.

Comment (ojaswi12) [reply doc/DISTUTILS.rst.txt]: yeah, I did, but I didn't push it yet

**Comment by mattip** [comment]
Replaced with #13729

**Pull request closed without merging**